### PR TITLE
msm8226: Initial support for Galaxy S3 Neo (s3ve3g)

### PR DESCRIPTION
**Pull request by luka177**

### CODE REVIEW

File: dts/msm8226/msm8226-samsung-ms01-r4.dts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id =  <0x9e00ff01 0x04 0x20000 0x9e08ff01 0x04 0x20000 0xdd08ff01 0x04 0x20000>;
```
Comment (TravMurav) [inline dts/msm8226/msm8226-samsung-ms01-r4.dts:10]:
```suggestion
	qcom,msm-id =  <0x9e00ff01 0x04 0x20000>, <0x9e08ff01 0x04 0x20000>, <0xdd08ff01 0x04 0x20000>;
```
and I assume you only need the second value for msm8226 so probably:
```suggestion
	qcom,msm-id =  <0x9e08ff01 0x04 0x20000>;
```

File: dts/msm8226/msm8226-samsung-ms01.dts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// This is used by the bootloader to find the correct DTB
+	qcom,msm-id =  <0x91 0x08 0x10001 0x9e 0x08 0x10001 0x9f 0x08 0x10001 0xc6 0x08 0x10001 0xcd 0x08 0x10001>;
```
Comment (TravMurav) [inline dts/msm8226/msm8226-samsung-ms01.dts:10]:
```suggestion
	qcom,msm-id =  <0x91 0x08 0x10001>, <0x9e 0x08 0x10001>, <0x9f 0x08 0x10001>, <0xc6 0x08 0x10001>, <0xcd 0x08 0x10001>;
```

but this looks like MTP which is not what samsung usually uses. Are you sure you need this file at all?

File: dts/msm8226/msm8226-samsung-ms01-r4.dts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	
+	ms01 {
+		model = "Samsung Galaxy S3 Neo";
+		compatible = "samsung,ms01", "qcom,msm8226", "lk2nd,device";
```
Comment (TravMurav) [inline dts/msm8226/msm8226-samsung-ms01-r4.dts:14]: Is there something in the device cmdline that could be used as a match rule here?

File: dts/msm8226/rules.mk [hunk 4]
```diff
@@ -9,6 +9,8 @@ DTBS += \
 	$(LOCAL_DIR)/apq8026-samsung-r09.dtb \
 	$(LOCAL_DIR)/msm8226-motorola-falcon.dtb \
 	$(LOCAL_DIR)/msm8226-samsung-ms013g.dtb \
+	$(LOCAL_DIR)/msm8226-samsung-ms01.dtb \
+       $(LOCAL_DIR)/msm8226-samsung-ms01-r4.dtb \
```
Comment (TravMurav) [inline dts/msm8226/rules.mk:13]: Please use tabs

File: dts/msm8226/msm8226-samsung-ms01.dts (hunk 2 above)
Comment (luka177) [reply dts/msm8226/msm8226-samsung-ms01.dts:10]: I am not sure how could i fix

Comment (TravMurav) [reply dts/msm8226/msm8226-samsung-ms01.dts:10]: drop the file and check if lk2nd still boots on the device?

Comment (luka177) [reply dts/msm8226/msm8226-samsung-ms01.dts:10]: Well it for sure will , but i thought those are 2 different revisions

Comment (TravMurav) [reply dts/msm8226/msm8226-samsung-ms01.dts:10]: Unless you can confirm that there is a device that really needs to match mtp to work, please drop. This really is not something samsung ever used in my experience, and is just from the default qcom code most likely

Comment (luka177) [reply dts/msm8226/msm8226-samsung-ms01.dts]: done

File: dts/msm8226/msm8226-samsung-ms01-r4.dts (hunk 3 above)
Comment (TravMurav) [reply dts/msm8226/msm8226-samsung-ms01-r4.dts]: I'd still appreciate having some match rule for this, could you share the cmdline you get in the lk log maybe?

**Pull request closed without merging**